platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/TypedAdditionalData.java

```diff
@@ -19,23 +19,21 @@
 import org.gradle.api.Incubating;
 import org.gradle.api.problems.AdditionalData;
 
-import java.util.Map;
-
 /**
- * General additional data type that can be used to attach arbitrary data to a problem with a string map.
+ * additional data type that can be used to attach arbitrary data to a problem using isolated serialized data.
  *
  * @since 8.13
  */
 @Incubating
 public interface TypedAdditionalData extends AdditionalData {
 
     /**
-     * provides data to reconstruct the object
+     * the serialized isolated additional data
      *
-     * @return data
-     * @since 8.13
+     * @return the data
+     * @since 8.14
```

Comment:
Why are we marking this interface with `@Incubating` and `@since` if it is internal? Is it intended to be public?

If this is intended to be public, I do not think we should be exposing the raw bytes representation of the object to the user.